### ISSUE TITLE
Maximizing an always on top transparent frameless window causes the bottom to overflow under the taskbar

### ISSUE THREAD

**Issue by Yonezpt** [nwjs/nw.js#6204]
NWJS Version : SDK 26.0 beta 1
Operating System : Win7 x64

### Expected behavior

The bottom of the window should not go under the taskbar when maximized.

### Actual behavior

The bottom of the window goes under the taskbar when maximized.

### How to reproduce

Basic app template with window `frame` set to `false`, `transparent` set to `true` and `always_on_top` set to `true`
Maximize app and check the bottom of the window under the windows taskbar

Resulting behavior illustrated:

![ss](https://user-images.githubusercontent.com/8498057/31525295-cbedd2f4-afae-11e7-8474-fabb49ad3a98.png)

Expected behavior:

![ss](https://user-images.githubusercontent.com/8498057/31525321-fd8a64f8-afae-11e7-8add-f6cc3c17f06c.png)

**Comment by rogerwang** [comment]
Please try this version:
https://dl.nwjs.io/live-build/10-06-2017/nw26-4fae369-edf758a-6de1a29-61e38b8

**Comment by Yonezpt** [comment]
@rogerwang The problem is still there and I just noticed that I failed to include another detail so that this issue can be reproduced. I have updated the main issue with the missing detail. The window also needs to be set to `always_on_top: true`. Without that, the problem does not exist.

So to summarize: Frameless, transparent and always on top causes this issue, even with the latest RC1 you just shared. Setting either one of those three options to false (or removing it) makes the window behave correctly.

**Comment by jtg-gg** [comment]
@Yonezpt can you sent us your sample app ?

**Comment by Yonezpt** [comment]
@jtg-gg Certainly, sorry for the delay. This sample app makes use of the `package.nw` folder method. The contents of that folder are the two files below:

`package.json`

````json
{
    "author": "test",
    "description": "test",
    "main": "index.html",
    "name": "test",
    "version": "1.0.0",
    "window": {
        "id": "test",
        "always_on_top": true,
        "frame": false,
        "transparent": true
    }
}
````

`index.html`

````html
<!DOCTYPE html>
<html>
    <head>
    </head>
    <body style="background: #e2e2e2">
        <button onclick="resize();return false;">Click to maximize/restore</button>
        <script>
                var is_maximized = false;
                var gui = require("nw.gui");
                var win = gui.Window.get();

                win.on("restore", () => is_maximized = false);
                win.on("maximize", () => is_maximized = true);

                function resize() {
                    if (is_maximized) {
                        win.restore();
                    } else {
                        win.maximize();
                    }
                }
        </script>
    </body>
</html>
````

**Comment by ubaidu15** [comment]
set frame to false and always on top to true, refresh it and compile again, or start with a new name of project and copy the code

**Comment by Yonezpt** [comment]
@ubaidu15 that is not a solution and it is not something I didn't already know:

> [Removing either one of those three options makes the window behave correctly when maximized.](https://github.com/nwjs/nw.js/issues/6204#issuecomment-336323158)

that's simply avoiding the problem which is not what I want, otherwise I would not have opened this issue at all if I didn't want a transparent frameless window with the option to stay always on top.

**Comment by Christywl** [comment]
I try the sample and can reproduce it on Windows 8.1 with nwjs-sdk-v0.25.4.
![image](https://user-images.githubusercontent.com/20039821/31596446-2e510e48-b275-11e7-92fa-e34a748f464c.png)

**Comment by rogerwang** [comment]
@jtg-gg thanks. Merged into nw26 master patch. Not merged into nw25 because there is no 0.25 release any more -- 0.26.0 will be released before tomorrow.
